pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -247,9 +247,15 @@ func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machi
 	}
 
 	if instance == nil {
-		// The machine hasn't been created yet
-		klog.Info("Instance is nil and therefore does not exist")
-		return nil
+		instance, err = ec2svc.InstanceByTags(scope)
+		if err != nil {
+			return errors.Errorf("failed to query instance by tags: %+v", err)
+		}
+		if instance == nil {
+			// The machine hasn't been created yet
+			klog.Info("Instance is nil and therefore does not exist")
```

Comment:
This should probably be behind a `.V(3)` or higher -- I'm trying to imagine a situation when I'm reading logs and I see this line pop up. I think it either needs more context or should not be in the Info log

Reply:
+1